content/en/docs/configuration/selfsigned.md

```diff
@@ -6,27 +6,29 @@ type: "docs"
 ---
 
 The `SelfSigned` issuer doesn't represent a certificate authority as such, but
-instead denotes that certificates will be signed through "self signing" using
-the given private key. This means that the provided private key of the resulting
-certificate will be used to sign its own certificate.
-
-This `Issuer` type is useful for bootstrapping the CA certificate key pair for
-some Private Key Infrastructure (PKI), or for otherwise creating simple
-certificates.  Clients consuming these certificates have _no way_ to trust this
-certificate since there is no CA signer apart from itself, and as such, would be
-forced to trust the certificate as is.
-
-> Note: `CertificateRequests` that reference a self signed certificate _must_
-> also contain the annotation `cert-manager.io/private-key-secret-name`. This is
-> because without access to the private key of the certificate request, the
-> `CertificateRequest` will be unable to self sign the certificate. This
-> annotation is added automatically by the `Certificate` controller.
+instead denotes that certificates will "sign themselves" using a given private
+key. In other words, the private key of the certificate will be used to sign
+the certificate itself.
+
+This `Issuer` type is useful for bootstrapping a root certificate for a
+custom PKI (Public Key Infrastructure), or for otherwise creating simple
+ad-hoc certificates.
```

Comment:
I think this change removes some of the intention of the original text, which is that the self signed issuer can be used to bootstrap a CA Issuer. Can we add that back in?


```yaml
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  name: selfsigned-issuer
spec:
  selfSigned: {}
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: osm-ca
  namespace: sandbox
spec:
  isCA: true
  commonName: osm-system
  secretName: osm-ca
  issuerRef:
    name: selfsigned-issuer
    kind: ClusterIssuer
    group: cert-manager.io
---
apiVersion: cert-manager.io/v1alpha2
kind: Issuer
metadata:
  name: osm-ca
  namespace: sandbox
spec:
  ca:
    secretName: osm-ca
```

Reply:
I would personally _never_ have gotten a hint about bootstrapping `CA` issuers using `SelfSigned` from the original document. I've read it over a few times now and I just can't see any suggestion of that. It talks generally about bootstrapping PKI, but I can't see any explicit link to `CA` issuers. Am I missing something?

It definitely seems like a really worthwhile thing to _add_ and a good use-case to highlight, though, including the config you posted.

Is there a reason the CA issuer in your example uses `apiVersion: cert-manager.io/v1alpha2` or am I OK to change it to `apiVersion: cert-manager.io/v1`?

Reply:
Yep, you're right, it wasn't even very well implied this is an intended and recommended way to bootstrap a CA Issuer.

Yes please change it to `v1`. Copy paste from some old manifest examples I had :smile: